COLLABORATOR_GUIDE.md

```diff
@@ -84,6 +87,197 @@ All pull requests that modify executable code should be subjected to
+changes to error handling may have been made. Additional CI testing may be
+required.
+
+#### When breaking changes actually break things
+
+Breaking changes are difficult primarily because they change the fundamental
+assumptions a user of Node.js has when writing their code and can cause
+existing code to stop functioning as expected -- costing developers and users
+time and energy to fix.
+
+Because breaking (semver-major) changes are permitted to land in master at any
+time, it should be *understood and expected* that at least some subset of the
+user ecosystem *may* be adversely affected *in the short term* when attempting
+to build and use Node.js directly from master. This potential instability is
+precisely why Node.js offers distinct Current and LTS release streams that
+offer explicit stability guarantees.
+
+Specifically:
+
+* Breaking changes should *never* land in Current or LTS except when:
+  * Resolving critical security issues.
+  * Fixing a critical bug (e.g. fixing a memory leak) requires a breaking
+    change.
+  * There is CTC consensus that the change is required.
+* If a breaking commit does accidentally land in a Current or LTS branch, an
+  attempt to fix the issue will be made before the next release; If no fix is
+  provided then the commit will be reverted.
+
+When any change is landed in master, and it is determined that the such
+changes *do* break existing code, a decision may be made to revert those
+changes either temporarily or permanently. However, the decision to revert or
+not can often be based on many complex factors that are not easily codified. It
+is also possible that the breaking commit can be labeled retroactively as a
+semver-major change that will not be backported to Current or LTS branches.
+
+### Deprecations
+
+Deprecation refers to the identification of Public APIs that should no longer
+be used and that may be removed or modified in non-backwards compatible ways in
+a future major release of Node.js. Deprecation *may* be used with internal APIs
+if there is expected impact on the user community.
+
+Node.js uses three fundamental Deprecation levels:
+
+* *Documentation-Only Deprecation* refers to elements of the Public API that are
+  being staged for deprecation in a future Node.js major release. An explicit
+  notice indicating the deprecated status is added to the API documentation
+  *but no functional changes are implemented in the code*. There will be no
+  runtime deprecation warning emitted for such deprecations.
+
+* *Runtime Deprecation* refers to the use of process warnings emitted at
+  runtime the first time that a deprecated API is used. A command-line
+  switch can be used to escalate such warnings into runtime errors that will
+  cause the Node.js process to exit. As with Documentation-Only Deprecation,
+  the documentation for the API must be updated to clearly indicate the
+  deprecated status.
+
+* *End-of-life* refers to APIs that have gone through Runtime Deprecation and
```

Comment:
They'll be removed from the main docs, but they will still be listed in the `deprecations.md` documentation page that lists the various deprecation identifiers.